### ISSUE TITLE
The ApiSelect remote is triggered each time when the value of another field changes, updating the parameters used for the API select.

### ISSUE THREAD

**Issue by doraemonxxx** [vbenjs/vue-vben-admin#3767]
**⚠️ IMPORTANT ⚠️ Please check the following list before proceeding. If you ignore this issue template, your issue will be directly closed.**

- [x] Read [the docs](https://anncwb.github.io/vue-vben-admin-doc/).
- [x] Make sure the code is up to date. (Some bugs have been fixed in the latest code)
- [x] This is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/anncwb/vue-vben-admin/discussions) or join our [Discord](https://discord.gg/8GuAdwDhj6) Chat Server.

### Describe the bug

The ApiSelect remote search should only be triggered when selecting an option, not when other input fields change the API parameter.

### Reproduction

Design a form with two input fields: a RadioGroup and an ApiSelect. When a value is chosen in the RadioGroup, it will update a reference variable and become the parameter value for the ApiSelect. However, please note that the current setup causes the ApiSelect's remote search/filter to activate whenever the parameters change. In simpler terms, selecting a value in the RadioGroup triggers the ApiSelect's search/filter mechanism.

```
watch(
    () => props.params,
    (value, oldValue) => {
      if (isEqual(value, oldValue)) return;
      fetch();
    },
    { deep: true, immediate: props.immediate },
  );

```
Please see screen video
https://github.com/vbenjs/vue-vben-admin/assets/88119911/675e8a02-f53b-4591-bdab-ff5e3de5e307

Reference revision:
ccf4027533d2adabd21bade025ca7bc7d34d75f6

## System Info

- Operating System: Windows 11
- Node version: v20.11.0
- Package manager (npm/yarn/pnpm) and version: pnpm 8.14.1

**Comment by electroluxcode** [comment]
> is there any way to replicate the bug of [ccf4027](https://github.com/vbenjs/vue-vben-admin/commit/ccf4027533d2adabd21bade025ca7bc7d34d75f6) ? and why the revision is implemented.

I found a commit message in the code. I think it should answer your question。
- fix(BasicForm->ApiRadioGroup): when options click, duplicate requests. resolve #3387 。

But I find this commit is unsatisfactory because duplicate requests should be controlled by themselves in the business

**Comment by electroluxcode** [comment]
> Hello there @electroluxcode
> 
> Here's an example of how to replicate the issue.
> 
> ```
> <template>
>   <PageWrapper title="Issue ApiSelect #3767">
>     <BasicForm @register="registerPreview" class="my-5">
>       <template #custom="{ model, field }">
>         <ApiSelect
>           :api="api"
>           :params="{ test: paramForApiSelect }"
>           v-model:value="model[field]"
>           resultField="data"
>           labelField="label"
>           valueField="value"
>         />
>       </template>
>     </BasicForm>
>   </PageWrapper>
> </template>
> <script lang="ts" setup>
>   import { BasicForm, useForm, ApiSelect } from '@/components/Form';
>   import { PageWrapper } from '@/components/Page';
>   import { ref } from 'vue';
> 
>   const paramForApiSelect = ref('init');
> 
>   let api: any = async (param: any) => {
>     return new Promise((rev) => {
>       console.log('trigger-api:', param);
>       rev([
>         { value: `11`, label: `11` },
>         { value: `22`, label: `22` },
>       ]);
>     });
>   };
> 
>   const [registerPreview] = useForm({
>     schemas: [
>       {
>         field: 'radio',
>         component: 'RadioGroup',
>         label: 'Radio',
>         colProps: {
>           span: 24,
>         },
>         componentProps: () => {
>           return {
>             options: [
>               {
>                 label: 'test-1',
>                 value: 'test-1',
>               },
>               {
>                 label: 'test-2',
>                 value: 'test-2',
>               },
>               {
>                 label: 'test-3',
>                 value: 'test-3',
>               },
>               {
>                 label: 'test-4',
>                 value: 'test-4',
>               },
>             ],
>             onChange: (e: { target: any }) => {
>               const target = e?.target;
>               if (target ?? false) {
>                 console.log('target.value', target.value);
>                 paramForApiSelect.value = target.value;
>               }
>             },
>           };
>         },
>       },
>       {
>         field: 'select',
>         label: 'ApiSelect',
>         colProps: { span: 4 },
>         slot: 'custom',
>       },
>     ],
>     actionColOptions: {
>       span: 18,
>     },
>   });
> </script>
> ```
> 
> When selecting a value in Radio, the ApiSelect API will triggered, because it watches the parameters and will fetch if the new value and old value are not the same. This shouldn't be the behavior right? Originally, the API should only triggered when you open an option in the ApiSelect menu, not when you simply select a value in the Radio.
> 
> By the way, I used a slot to customize the ApiSelect. In my local development, I added extra information and stylings to the options.
> 
> Thanks!

I think there might be something wrong with your use case. The param and api in ApiSearch should be designed for remote search. If you revert this commit, you'll find that remote searches can't trigger the api because isFirstLoaded keeps returning true.

**Comment by electroluxcode** [comment]
As a  example, you can enter some fields in the ApiSelect component to try to trigger the api. You can try revert before and after to valiate

```vue
<template>
  <PageWrapper title="issue ApiRadioGroup">
    <BasicForm @register="registerPreview" class="my-5">
      <template #custom="{ model, field }">
        <ApiSelect
          :api="api"
          @Search="mySearch"
          :params="{ test: paramForApiSelect }"
          v-model:value="model[field]"
          resultField="data"
          labelField="label"
          valueField="value"
          :showSearch="true"
        />
      </template>
    </BasicForm>
  </PageWrapper>
</template>
<script lang="ts" setup>
  import { BasicForm, useForm, ApiSelect } from '@/components/Form';
  import { PageWrapper } from '@/components/Page';
  import { ref } from 'vue';
  const paramForApiSelect = ref('init');
  const mySearch = (v)=>{
    paramForApiSelect.value = v
    console.log("mySearch")
  }
  let api: any = async (param: any) => {
    return new Promise((rev) => {
      console.log('trigger-api:', param);
      rev([
        { value: `11`, label: `11` },
        { value: `22`, label: `22` },
      ]);
    });
  };
  const [registerPreview] = useForm({
    schemas: [
      {
        field: 'select',
        label: 'ApiSelect',
        colProps: { span: 4 },
        slot: 'custom',
      },
    ],
    actionColOptions: {
      span: 18,
    },
  });
</script>

```
